src/collector.ts

```diff
@@ -152,16 +189,24 @@ class NaiveGomodParser {
 
             if (exactMatchDep) {
                 // Software stack uses the module
+                let replaceDependency = checkReplaceDependency(exactMatchDep);
+                if (replaceDependency) {
```

Comment:
This can be avoided if you implement previous comment of returning same deps if no replacement is found.